client/src/Widgets/ActivityTable.tsx

```diff
@@ -0,0 +1,277 @@
+  Editable,
+  EditableInput,
+  EditablePreview,
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Td,
+  Tbody,
+  Box,
+  HStack,
+  Menu,
+  MenuButton,
+  MenuList,
+  Show,
+} from "@chakra-ui/react";
+import { GoKebabVertical } from "react-icons/go";
+import { FaFolder } from "react-icons/fa";
+import { RiDraftFill } from "react-icons/ri";
+import { MdAssignment } from "react-icons/md";
+import { BsPeopleFill } from "react-icons/bs";
+import { useFetcher, useNavigate } from "react-router-dom";
+import { AssignmentStatus } from "../Tools/_framework/Paths/ActivityEditor";
+
+export default forwardRef(function ActivityTable(
+  {
+    content,
+    suppressAvatar,
+    showOwnerName,
+    showAssignmentStatus,
+    showPublicStatus,
+  }: {
+    content: {
+      cardLink?: string;
+      id: number;
+      assignmentStatus?: AssignmentStatus;
+      isFolder?: boolean;
+      isPublic?: boolean;
+      isShared?: boolean;
+      title: string;
+      ownerName?: string;
+      menuItems?: ReactElement;
+      editableTitle?: boolean;
+      autoFocusTitle?: boolean;
+      closeTime?: string;
+      authorRow?: boolean;
+    }[];
+    suppressAvatar?: boolean;
+    showOwnerName?: boolean;
+    showAssignmentStatus?: boolean;
+    showPublicStatus?: boolean;
+  },
+  ref: React.ForwardedRef<HTMLButtonElement>,
+) {
+  const navigate = useNavigate();
+  const fetcher = useFetcher();
+
+  // this state does not do much; its purpose is to prevent the edge case of someone clicking to select the text input, and,
+  // without letting go, dragging it on top of a link so that the link is triggered
+  const [titleBeingEdited, setTitleBeingEdited] = useState(false);
+
+  return (
+    <TableContainer>
+      <Table>
+        <Thead>
+          <Tr borderBottom="2px solid gray">
+            <Th></Th>
+            <Th></Th>
+            {showPublicStatus ? <Th>Visibility</Th> : null}
+            <Show above="md">
+              {" "}
+              {showAssignmentStatus ? <Th>Assignment Status</Th> : null}{" "}
+            </Show>
+            {showOwnerName || !suppressAvatar ? <Th>Owner</Th> : null}
+            <Th></Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {content.map(function (activity) {
+            let assignmentStatusString: string =
+              activity.assignmentStatus !== null &&
+              activity.assignmentStatus !== undefined &&
+              activity.assignmentStatus !== "Unassigned"
+                ? activity.assignmentStatus
+                : "";
+
+            if (
+              activity.assignmentStatus === "Open" &&
+              activity.closeTime !== undefined
+            ) {
+              assignmentStatusString =
+                assignmentStatusString + " until " + activity.closeTime;
+            }
+
+            function saveUpdatedTitle(newTitle: string) {
+              if (newTitle !== activity.title && activity.id !== undefined) {
+                fetcher.submit(
```

Comment:
Don't let the new title be a blank "" or you won't be able to change it back here, due to the `<Editable>` being too small. Instead, if `newTitle===""`, then just set the name to "Untitled".